ui/src/time_picker.tsx

```diff
@@ -0,0 +1,269 @@
+              },
+              text: {
+                primary: fluentPalette.neutralPrimary,
+                secondary: cssVar('$neutralSecondary'),
+                disabled: cssVar('$neutralTertiaryAlt'),
+              },
+              action: {
+                active: fluentPalette.themePrimary,
+                disabled: cssVar('$neutralLight'),
+                hover: cssVar('$neutralLight'),
+                hoverOpacity: 0.04
+              }
+            },
+          }
+        ))
+      }),
+      [AdapterDateFns, setAdapterDateFns] = React.useState<typeof DateFnsUtils | null>(),
+      getAdapterDateFns = () => import('@mui/x-date-pickers/AdapterDateFns').then(({ AdapterDateFns }) => { setAdapterDateFns(() => AdapterDateFns) }),
+      textInputProps = {
+        onClick: () => setIsDialogOpen(true),
+        placeholder: m.placeholder,
+        value,
+        label: m.label,
+        required: m.required,
+        hour_cycle: hour_cycle,
+        disabled: m.disabled,
+        errorMsg: `Wrong input. Please enter the time in range from 
+          ${formatDateToTimeString(parseTimeStringToDate(m.min || '00:00'), hour_cycle)} to 
+          ${formatDateToTimeString(parseTimeStringToDate(m.max || '00:00'), hour_cycle)}.
+        `
+      }
+
+    React.useEffect(() => {
+      wave.args[m.name] = defaultVal ? formatDateToTimeString(defaultVal, '24') : null
+      getTheme()
+      getAdapterDateFns()
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    React.useEffect(() => {
+      if (theme) {
+        theme.palette.primary.main = fluentPalette.themePrimary
+        theme.palette.text.primary = fluentPalette.neutralPrimary
+        theme.palette.action.active = fluentPalette.themePrimary
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [fluentPalette.themePrimary, theme])
+
+    return <>
+      <React.Suspense fallback={<LazyLoadPlaceholder />}>
+        {theme && AdapterDateFns
+          ? <ThemeProvider theme={theme}>
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+              <TimePicker
+                value={value}
+                label={m.label}
+                open={isDialogOpen}
+                onChange={(value, _keyboardInputValue) => setValue(value as Date)}
+                onAccept={(value) => onSelectTime(value as Date)}
+                onClose={() => setIsDialogOpen(false)}
+                ampm={hour_cycle === '12'}
+                showToolbar={true}
+                ToolbarComponent={params => <Toolbar params={params} label={m.label} switchAmPm={switchAmPm} />}
+                PopperProps={{ anchorEl: () => textInputRef.current as VirtualElement, ...popoverProps }}
+                minTime={m.min ? parseTimeStringToDate(m.min) : undefined}
+                maxTime={m.max ? parseTimeStringToDate(m.max) : undefined}
+                minutesStep={[1, 5, 10, 15, 20, 30, 60].includes(m.minutes_step || 1) ? m.minutes_step : 1}
```

Comment:
```suggestion
                minutesStep={allowedMinutesSteps[minutes_step]}
```

where `allowedMinutesSteps` is a constant:

```ts
  allowedMinutesSteps: { [key: U]: U } = { 1: 1, 5: 5, 10: 10, 15: 15, 20: 20, 30: 30, 60: 60 }
```